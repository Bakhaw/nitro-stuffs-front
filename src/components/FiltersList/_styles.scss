.FiltersList {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  margin-top: 80px;

  &__Title {
    color: $White;
  }

  &__Filters {
    margin-top: 20px;
  }
}
