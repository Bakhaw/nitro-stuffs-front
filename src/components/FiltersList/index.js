import React from 'react';
import PropTypes from 'prop-types';

import Filter from '../Filter';

import filtersList from './filters';

function FilterList({ activeFilters, handleFilterClick }) {
  return (
    <div className='FiltersList'>
      <h3 className='FiltersList__Title'>FILTERS</h3>
      <ul className='FiltersList__Filters'>
        {filtersList.map((filter) => (
          <li key={filter.type}>
            <Filter
              active={activeFilters.includes(filter.type)}
              filter={filter}
              handleFilterClick={handleFilterClick}
            />
          </li>
        ))}
      </ul>
    </div>
  );
}

FilterList.propTypes = {
  activeFilters: PropTypes.array.isRequired,
  handleFilterClick: PropTypes.func.isRequired,
};

export default FilterList;
