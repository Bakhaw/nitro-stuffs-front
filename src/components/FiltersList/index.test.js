import React from 'react';
import { shallow } from 'enzyme';

import FiltersList from '.';

test('should render properly', () => {
  const props = {
    activeFilters: ['smoke'],
    handleFilterClick: jest.fn(),
  };

  const container = shallow(<FiltersList {...props} />);
  expect(container.debug()).toMatchSnapshot();
});
