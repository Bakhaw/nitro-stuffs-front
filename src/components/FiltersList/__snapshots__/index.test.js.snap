// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div className=\\"FiltersList\\">
  <h3 className=\\"FiltersList__Title\\">
    FILTERS
  </h3>
  <ul className=\\"FiltersList__Filters\\">
    <li>
      <Filter active={true} filter={{...}} handleFilterClick={[Function: mockConstructor]} />
    </li>
    <li>
      <Filter active={false} filter={{...}} handleFilterClick={[Function: mockConstructor]} />
    </li>
    <li>
      <Filter active={false} filter={{...}} handleFilterClick={[Function: mockConstructor]} />
    </li>
  </ul>
</div>"
`;
